web/Web/web/CAL/views.py

```diff
@@ -198,3 +199,70 @@ def get_ajax(self, request, *args, **kwargs):
         except CALError as e:
             # TODO: add proper http response for CAL errors
             return JsonResponse({"message": "Ops! CALError."}, status=404)
+
+
+class DSLoggingView(views.CsrfExemptMixin,
+                    views.LoginRequiredMixin,
+                    RetrievalMethodPermissionMixin,
+                    views.JsonRequestResponseMixin,
+                    generic.View):
+    """
+    View to log info of Dynamic Sampling
+    """
+    require_json = False
+
+    def post(self, request, *args, **kwargs):
+        session = self.request.user.current_session.uuid
+
+        try:
+            stratum_num = self.request_json.get(u"stratum_number")
+            stratum_size = self.request_json.get(u"stratum_size")
+            T = self.request_json.get(u"T")
+            N = self.request_json.get(u"N")
+            R = self.request_json.get(u"R")
+            current_sample_size = self.request_json.get(u"n")
+
+            # Get docs in current stratum
+            docs = CALFunctions.get_stratum_documents(str(session))
+            # Get sampled docs
+            sampled_docs, _ = CALFunctions.get_documents(str(session), -1)
+
+        except KeyError:
+            error_dict = {u"message": u"Errors when getting DS info."}
+            return self.render_bad_request_response(error_dict)
+
+        if not docs:
+            return self.render_bad_request_response({u"message": u"Empty set for stratum {}".format(stratum_num)})
+
+        if not sampled_docs:
+            return self.render_bad_request_response({u"message": u"Empty sample for stratum {}".format(stratum_num)})
+
+        if int(stratum_num) == 1:
+            current_sample_size = 1
+
+        docs_list = {}
+        for docid_score_pair in docs:
+            doc_id, score = docid_score_pair.rsplit(':', 1)
+            docs_list[doc_id] = float(score)
+
+        sampled_docs_list = []
+        for docid_score_pair in sampled_docs:
+            doc_id, score = docid_score_pair.rsplit(':', 1)
+            sampled_docs_list.append(doc_id)
+
+        exists = DS_logging.objects.filter(user=self.request.user,
+                                           stratum_num=stratum_num,
+                                           session=self.request.user.current_session)
+
+        if exists:
+            return self.render_bad_request_response({u"message": u"Duplicate record for stratum {}".format(stratum_num)})
+        else:
+            record = {'user': self.request.user, 'session': self.request.user.current_session,
+                      'stratum_size': int(stratum_size), 'stratum_num': int(stratum_num),
+                      'sample_size': int(current_sample_size),
+                      'T': int(T), 'N': int(N), 'R': int(R), 'stratum_docs': json.dumps(docs_list),
+                      'sampled_docs': json.dumps(sampled_docs_list)}
+            DS_logging.objects.create(**record)
```

Comment:
Just specify the parameters in the `create` function instead of having to create `record`. 

```
DS_logging.objects.create(user = self.request.user,
                          session = self.request.user.current_session,
....
```

Reply:
Fixed